pkgs/development/python-modules/ddddocr/default.nix

```diff
@@ -0,0 +1,45 @@
+{
+  lib,
+  stdenv,
+
+  buildPythonPackage,
+  fetchPypi,
+  setuptools,
+  numpy,
+  onnxruntime,
+  pillow,
+  opencv-python-headless,
+}:
+
+buildPythonPackage rec {
+  pname = "ddddocr";
+  version = "1.5.6";
+  pyproject = true;
+
+  # The maintainers do not track versions as tags or releases in GitHub, only
+  # in PyPI - so it's preferable to use fetchPypi.
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-KDmpQL+r4C4yhO8/nSoDcpKqn2QfNVtDqbcL7Onhtz0=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    numpy
+    onnxruntime
+    pillow
+    opencv-python-headless
+  ];
+
+  # https://github.com/microsoft/onnxruntime/issues/10038 - in the restricted
+  # build environment, ARM64 cannot import ddddocr (but it can normally).
+  pythonImportsCheck = lib.optionals (!stdenv.isAarch64) ["ddddocr"];
+
+  meta = {
+    description = "OCR library trained against simple CAPTCHAs for Latin and Chinese characters";
+    homepage = "https://github.com/sml2h3/ddddocr";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ simonhollingshead ];
```

Comment:
you'll probably want to add a platforms here instead and just set the pythonImports check to the list so that hydra does not build the package for other platforms and people try to use it (then fail).

Reply:
The package works fine when not run inside a limited build container (i.e. when /sys/devices is readable at runtime).

It seems that pytorch cpuinfo provides a default value for x86 when it's unreadable, but an error for ARM when it's unreadable (fun!).

onnxruntime has to turn off a lot of tests in the build container for the same reason.

https://github.com/NixOS/nixpkgs/blob/8ef82d9b1dd1df062acb74eba5928738d951facb/pkgs/by-name/on/onnxruntime/package.nix#L250